YellowNotes.Api/Controllers/UserController.cs

```diff
@@ -0,0 +1,83 @@
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using YellowNotes.Core.Dtos;
+using YellowNotes.Core.Services;
+using YellowNotes.Core.Utility;
+using System.Threading;
+
+namespace YellowNotes.Api.Controllers
+{
+    [Authorize]
+    [ApiController]
+    [Route("users")]
+    public class UserController : ControllerBase
+    {
+        private readonly IUserService userService;
+
+        public UserController(IUserService userService) => this.userService = userService;
+
+        [AllowAnonymous]
+        [HttpPost("register")]
+        public async Task<IActionResult> Register([FromBody] UserDto userDto,
+            CancellationToken cancellationToken = default)
+        {
+            var success = await userService.CreateUser(userDto, cancellationToken);
+            if (!success)
+            {
+                return BadRequest("User cannot be created");
+            }
+
+            var token = userService.GenerateJWT(userDto);
+            return Ok(new { token });
+        }
+
+        [AllowAnonymous]
+        [HttpPost("authenticate")]
+        public async Task<IActionResult> Authenticate([FromBody] UserDto userDto,
+            CancellationToken cancellationToken = default)
+        {
+            var success = await userService.VerifyPassword(userDto, cancellationToken);
+            if (!success)
+            {
+                return BadRequest("Verification has failed");
+            }
+
+            var token = userService.GenerateJWT(userDto);
+            return Ok(new { token });
+        }
+
+        [HttpGet]
+        public IActionResult GetSomeTestContent([FromBody] UserDto userDto)
```

Comment:
what is that?

Reply:
A testing method, not needed for production and will be removed.